tests/integration/ceph_mgr_test.go

```diff
@@ -177,7 +177,7 @@ func (s *CephMgrSuite) waitForOrchestrationModule() {
 			// Get status information
 			bytes := []byte(output)
 			var status orchStatus
-			err := json.Unmarshal(bytes, &status)
+			err := json.Unmarshal(bytes[:len(output)], &status)
```

Comment:
@jmolmo How does this change the behavior? By definition the length of `bytes` should be the same as `len(output)`, so I don't see what this fixes.

Reply:
I suspect that the "output" string  sometimes contains more hidden characters, at least two more "." (points). Why ? no idea...  because apparently the output reported is 82 chars long and we cannot see nothing estrange at the end of the string: 

`[{"addr": "fv-az120-523", "hostname": "fv-az120-523", "labels": [], "status": ""}]`

Surprisingly, the error we catch in the log says that the string is 84 chars long ( SUPER WEIRD )

`Expected nil, but got: &json.SyntaxError{msg:"invalid character '.' after top-level value", Offset:84}`

I have tested  "Unmarshal" manually with the same output where we have the error ,and it works perfectly.  But if you modify the "output" adding a couple of points at the end, we have exactly the same error reported:

```
output length =  84   
invalid character '.' after top-level value
[{"addr": "fv-az120-523", "hostname": "fv-az120-523", "labels": [], "status": ""}]..
[]
```

Test it by yourself:  https://play.golang.org/p/QqoJDV5Zjam

So my suspicion is that in some way and sometimes (error appears from time to time) we have hidden chars at the end of the string. No idea about the root cause.. but is something that [also happens using string from other sources](https://stackoverflow.com/questions/25187718/invalid-character-x00-after-top-level-value).

I have used a "workaround". That is to use the detected length of the string (what we see and get from the orch command)  in the conversion of the string to a Bytes slice.

Reply:
In your test app, I see the same results whether unmarshaling `hosts[:len(output)]` or just `hosts`. If there is a `.` at the end of the output, it's the same error. If I remove the `.` from the end of the output, then it succeeds. So are you saying something will change by unmarshaling `bytes[:len(output)]` instead of just `bytes`? They still look equivalent to me.

Reply:
yes, that is the normal behavior, and I do not know how to simulate a "string" with more characters than the length reported. and that is the problem i think is happening .. we have a string that is represented properly but internally has more length than the one you get using the _len_ command. Is the only explanation i have.
In my view if you are not setting explicitly the length of the bytes slice, a couple more of chars are included as part of the string.

Reply:
Per discussion, let's add more logging such as for the length of the output so we can troubleshoot better if/when it happens again.